src/vs/editor/contrib/codeAction/browser/codeActionUi.ts

```diff
@@ -31,16 +31,18 @@ export class CodeActionUi extends Disposable {
 		quickFixActionId: string,
 		preferredFixActionId: string,
 		private readonly delegate: {
-			applyCodeAction: (action: CodeActionItem, regtriggerAfterApply: boolean) => Promise<void>;
+			applyCodeAction: (action: CodeActionItem, regtriggerAfterApply: boolean, preview: boolean) => Promise<void>;
 		},
 		@IInstantiationService instantiationService: IInstantiationService,
 	) {
 		super();
 
 		this._codeActionWidget = new Lazy(() => {
 			return this._register(instantiationService.createInstance(CodeActionMenu, this._editor, {
-				onSelectCodeAction: async (action) => {
-					this.delegate.applyCodeAction(action, /* retrigger */ true);
+				onSelectCodeAction: async (action, trigger) => {
+					if (trigger.preview !== undefined) {
```

Comment:
I think this may be incorrect as we always want to apply the refactoring . Instead try using `Boolean(trigger.preview)` to convert `trigger.preview` to a boolean